File: tools/build_standalone.py
import os
import subprocess
import re
import glob

base = r'C:\Users\kyzha\.openclaw\projects\zhan\code'
artifacts_dir = r'C:\Users\kyzha\.openclaw\projects\zhan\artifacts'
repo_dir = r'C:\Users\kyzha\.openclaw\projects\zhan'

# --- Determine version ---
# MAJOR: extract from existing artifacts (e.g. zhan_v1.9.html → 1)
major = 1
existing = glob.glob(os.path.join(artifacts_dir, 'zhan_v*.html'))
if existing:
    versions = []
    for f in existing:
        m = re.search(r'zhan_v(\d+)\.(\d+)\.html', os.path.basename(f))
        if m:
            versions.append((int(m.group(1)), int(m.group(2))))
    if versions:
        major = max(v[0] for v in versions)

# COMMIT_COUNT: from git
try:
    result = subprocess.run(
        ['git', 'rev-list', '--count', 'HEAD'],
        cwd=repo_dir,
        capture_output=True, text=True
    )
    commit_count = int(result.stdout.strip())
except Exception:
    commit_count = 0

version_str = f'v{major}.{commit_count}'
out = os.path.join(artifacts_dir, f'zhan_{version_str}.html')

# --- Build ---
with open(os.path.join(base, 'index.html'), 'r', encoding='utf-8') as f:
    html = f.read()

for name in ['style.css', 'relic.css']:
    with open(os.path.join(base, name), 'r', encoding='utf-8') as f:
        html = html.replace(
            f'<link rel="stylesheet" href="{name}">',
            f'<style>\n{f.read()}\n</style>'
        )

for name in ['data.js', 'core.js', 'ui.js']:
    with open(os.path.join(base, name), 'r', encoding='utf-8') as f:
        html = html.replace(
            f'<script src="{name}"></script>',
            f'<script>\n{f.read()}\n</script>'
        )

with open(out, 'w', encoding='utf-8') as f:
    f.write(html)

print(f'{version_str}: {os.path.getsize(out)} bytes')
print(f'OUTPUT: {out}')

File: tools/build_standalone.js
// build_standalone.js — 从源文件构建 zhan_standalone.html
// 内联 data.js + core.js + ui.js 到 index.html
var fs = require('fs');
var path = require('path');
var dir = path.resolve(__dirname, '..', 'code');

var dataJS = fs.readFileSync(dir + '/data.js', 'utf8');
var coreJS = fs.readFileSync(dir + '/core.js', 'utf8');
var uiJS = fs.readFileSync(dir + '/ui.js', 'utf8');
var html = fs.readFileSync(dir + '/index.html', 'utf8');

html = html.replace('<script src="data.js"></script>', '<script>\n' + dataJS + '\n</script>');
html = html.replace('<script src="core.js"></script>', '<script>\n' + coreJS + '\n</script>');
html = html.replace('<script src="ui.js"></script>', '<script>\n' + uiJS + '\n</script>');

// 标题改为斩 v14
html = html.replace('<title>斩 v10.0</title>', '<title>斩 v14</title>');

fs.writeFileSync(dir + '/zhan_standalone.html', html, 'utf8');
console.log('✅ zhan_standalone.html (' + html.length + ' bytes)');
